ethcore/sync/src/chain/supplier.rs

```diff
@@ -182,6 +183,10 @@ impl SyncSupplier {
 			} else if let Some(hdr) = io.chain().block_header(BlockId::Number(number)) {
 				data.append(&mut hdr.into_inner());
 				count += 1;
+				// Check that the packet won't be oversized
+				if data.len() > payload_soft_limit {
```

Comment:
Why not sum up the raw header bytes that we append to RLP a line earlier?

Reply:
Do you think it would be more efficient?

Reply:
Nitpick: this won't include the size of the fork header. we can move this to just before the `if reverse`.